src/theory/arith/callbacks.cpp

```diff
@@ -87,37 +87,25 @@ void FarkasConflictBuilder::reset(){
   d_consequent = NullConstraint;
   d_constraints.clear();
   d_consequentSet = false;
-  PROOF(d_farkas.clear());
```

Comment:
Ah, I see now that I should have been more clear earlier about how arith records farkas coefficients, and what we need to keep/remove.

Prior to this PR, arith maintained farkas coefficients in two locations:
a) In data structures in the `src/theory/arith` directory (e.g. `FarkasConflictBuilder` and `ConstraintRule`).
b) In data structures in the `src/proof` directory (e.g. `ArithProofRecorder`).

The former may be context-dependent. The latter were persistent.

We can remove the latter. They're replace by eager proof generators, etc., in the new proof system. **We cannot remove the former**: the new system uses them.

Generally, in the core of arithmetic, this means that we should treat proofs as "always on" and not trim code that is guarded by compile-time proof flags. Thus, this command should be kept, just `d_farkas.clear()`.

On the other hand, we can remove all of `src/proof/arith_*`, and direct references to such machinery in `src/theory/arith/`.

Reply:
I see, thanks! At first I had put most of this code under `#if 0 ... #endif` statements because it looked important. :) I have applied the changes you suggested, PTAL.